Patches/Seventh Seraph/7S_Research.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+	<Operation Class="PatchOperationFindMod">
+		<mods>
+		<li>Seventh Seraph Weapons</li>
+		</mods>
+		<match Class="PatchOperationSequence">
+		<operations>
+
+			<!-- ==================== Weapons research prerequisite patches ==================== -->
+
+
+			<li Class="PatchOperationReplace">
+				<xpath>/Defs/ThingDef[defName="SG_SeventhSeraphCQC"]/recipeMaker/researchPrerequisite</xpath>
+				<value>
+					<researchPrerequisite>ChargedShot</researchPrerequisite>
```

Comment:
You should be able to combine these into the CEMakeGunCompat operation. As reference, here's an operation from a random integrated patch that does so:
```
        <li Class="CombatExtended.PatchOperationMakeGunCECompatible">
          <defName>ARL_HMG</defName>
          <statBases>
            <WorkToMake>50000</WorkToMake>
            <Bulk>14.05</Bulk>
            <Mass>10.5</Mass>
            <SightsEfficiency>1</SightsEfficiency>
            <ShotSpread>0.06</ShotSpread>
            <SwayFactor>1.51</SwayFactor>
            <RangedWeapon_Cooldown>0.56</RangedWeapon_Cooldown>
          </statBases>
          <Properties>
            <recoilAmount>1.33</recoilAmount>
            <verbClass>CombatExtended.Verb_ShootCE</verbClass>
            <hasStandardCommand>true</hasStandardCommand>
            <defaultProjectile>Bullet_762x54mmR_FMJ</defaultProjectile>
            <burstShotCount>18</burstShotCount>
            <ticksBetweenBurstShots>6</ticksBetweenBurstShots>
            <warmupTime>1.25</warmupTime>
            <range>60</range>
            <soundCast>Shot_Minigun</soundCast>
            <soundCastTail>GunTail_Light</soundCastTail>
            <muzzleFlashScale>9</muzzleFlashScale>
          </Properties>
          <AmmoUser>
            <magazineSize>36</magazineSize>
            <reloadTime>3.33</reloadTime>
            <ammoSet>AmmoSet_762x54mmR</ammoSet>
          </AmmoUser>
          <FireModes />
          <weaponTags>
            <li>CE_Machinegun</li>
          </weaponTags>
          <researchPrerequisite>GasOperation</researchPrerequisite>
        </li>
        ```